xdsl/interpreters/experimental/pdl.py

```diff
@@ -0,0 +1,297 @@
+from dataclasses import (dataclass, field)
+from typing import Any
+
+from xdsl.ir import (Attribute, MLContext, MLIRType, OpResult, Operation,
+                     SSAValue)
+from xdsl.dialects import pdl
+from xdsl.dialects.builtin import IntegerAttr, IntegerType, ModuleOp
+from xdsl.pattern_rewriter import (PatternRewriter, PatternRewriteWalker,
+                                   AnonymousRewritePattern)
+from xdsl.interpreter import (Interpreter, InterpreterFunctions,
+                              register_impls, impl)
+from xdsl.utils.exceptions import InterpretationError
+from xdsl.utils.hints import isa
+
+
+@dataclass
+class PDLMatcher:
```

Comment:
I would like documentation here, because it's unclear how this works.
Since it's experimental, it's fine, but right now that means that it would take me significant time to modify anything if there is a bug, since there is no documentation.